src/metabase/version/task/upgrade_checks.clj

```diff
@@ -46,14 +46,23 @@
 (def ^:private job-key     "metabase.task.upgrade-checks.job")
 (def ^:private trigger-key "metabase.task.upgrade-checks.trigger")
 
+(defn- rand-hours
+  "Give a random hour plus the hour 12 hours away, i.e. one of [0 12], [1 13], [2 14], etc"
+  []
+  (let [hour-1 (rand-int 24)
+        hour-2 (mod (+ hour-1 12) 24)]
+    [hour-1 hour-2]))
+
 (defmethod task/init! ::CheckForNewVersions [_]
-  (let [job     (jobs/build
+  (let [[rand-hour-1 rand-hour-2] (rand-hours)
+        rand-minute (rand-int 60)
+        job     (jobs/build
                  (jobs/of-type CheckForNewVersions)
                  (jobs/with-identity (jobs/key job-key)))
         trigger (triggers/build
                  (triggers/with-identity (triggers/key trigger-key))
                  (triggers/start-now)
                  (triggers/with-schedule
                    ;; run twice a day
-                  (cron/cron-schedule "0 15 6,18 * * ? *")))]
+                  (cron/cron-schedule (format "0 %d %d,%d * * ? *" rand-minute rand-hour-1 rand-hour-2))))]
```

Comment:
Is there a way to add a test for this behavior?

Reply:
oh shoot, i hit merge before I saw this comment 😭 sorry

Reply:
I originally wrote a test for `rand-hours` but ended up deciding it was too silly to run it every time - either it would duplicate the exact logic of `rand-hours` itself, or have to be generative "run `rand-hours` 1000 times and assert that they're always 12 apart and under 24. Either way felt pretty silly.

I don't _think_ there's a good way to test the actual cron scheduling itself...